### PR TITLE
Feature news card

### PR DESCRIPTION
**Pull request by NirzanMalla**
Fetched News Api using axios, learned about the concepts of useEffect hook. The home page shows the news along with an image ,  news title and description. I will be adding more features in future such as ReadMore button and share button doing their specific work and the news displayed in home page is generic. In upcoming days, I will be working on feature which enables users to filter news based on their preferences.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+import Fone from "./pages/F1"
+import Golf from "./pages/Golf"
+import Nba from "./pages/Nba"
+import Ufc from "./pages/Ufc"
+import Swimming from "./pages/Swimming";
+import RootLayout from "./layout/RootLayout";
+
+const router = createBrowserRouter(
+  createRoutesFromElements(
+    <Route path="/" element={<RootLayout/>}>
```
Comment (subhashishjs) [inline src/App.jsx]: Don't use this name RootLayout. It's confusing. you can use Layout

File: src/components/NewsCard.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    typography:{
+      fontFamily: 'Poppins',
+      fontWeightLight: 200,
+      fontWeightRegular: 300,
+      fontWeightMedium: 400,
+      fontWeightBold: 500,
+    }
+  })
+
+const news = [
```
Comment (subhashishjs) [inline src/components/NewsCard.jsx]: keep this thing in seperate file. make something like data.js and export it from there

File: src/components/Stories.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import CardMedia from '@mui/material/CardMedia';
+import Typography from '@mui/material/Typography';
+import { Container, Grid } from '@mui/material';
+
+const Stories = () => {
+  const [data, setData] = useState([]);
+
+  const getNews = async () => {
+    try {
+      const response = await axios.get(
```
Comment (subhashishjs) [inline src/components/Stories.jsx]: now start using .env file. Always keep your api keys inside .env

**Review by subhashishjs — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (NirzanMalla) [reply src/App.jsx]: Done

File: src/components/NewsCard.jsx (hunk 2 above)
Comment (NirzanMalla) [reply src/components/NewsCard.jsx]: Done

File: src/components/Stories.jsx (hunk 3 above)
Comment (NirzanMalla) [reply src/components/Stories.jsx]: Done

File: src/App.jsx [hunk 4]
```diff
@@ -1,41 +1,39 @@
...
-        <Route path="/football" element={<Football/>}/>
-        <Route path="/f1" element={<Fone/>}/>
-        <Route path="/golf" element={<Golf/>}/>
-        <Route path="/nba" element={<Nba/>}/>
-        <Route path="/ufc" element={<Ufc/>}/>
-        <Route path="/swimming" element={<Swimming/>}/>
-      </Route>
+    <Route path="/" element={<Layout />}>
+      <Route exact path="/" element={<Home />} />
+      <Route path="/home" element={<Home />} />
```
Comment (subhashishjs) [inline src/App.jsx]: why two path are pointing towards the same page?

File: src/components/NewsCard.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        sx={{ height: 170 }}
+        image={
+          value.urlToImage ||
+          "https://static.vecteezy.com/system/resources/thumbnails/004/216/831/original/3d-world-news-background-loop-free-video.jpg"
+        }
+        title={value.title || "default image"}
+      />
+
+      <CardContent>
+        <Typography gutterBottom variant="h5" component="div">
```
Comment (subhashishjs) [inline src/components/NewsCard.jsx:31]: always use ? operator while doing these values. value?.description. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: src/pages/Football.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+const Football = () => {
+  const [data, setData] = useState([]);
+  const apikey = import.meta.env.VITE_API_KEY;
+  const getNews = async () => {
+    try {
+      const response = await axios.get(
+        `https://newsapi.org/v2/everything?q=football&apiKey=${apikey}`
+      );
+      setData(response.data.articles);
```
Comment (subhashishjs) [inline src/pages/Football.jsx]: use optional chaining here as well response?.data?.articles

**Review by subhashishjs — APPROVED** [review]

File: src/App.jsx (hunk 4 above)
Comment (NirzanMalla) [reply src/App.jsx]: Done

File: src/components/NewsCard.jsx (hunk 5 above)
Comment (NirzanMalla) [reply src/components/NewsCard.jsx:31]: done

File: src/pages/Football.jsx (hunk 6 above)
Comment (NirzanMalla) [reply src/pages/Football.jsx]: Done

**Pull request merged**